app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -51,26 +51,34 @@ class Hawkular::MiddlewareManager < ManageIQ::Providers::MiddlewareManager
     attr_accessor :client
 
     def verify_credentials(_auth_type = nil, options = {})
-      begin
-        # As the connect will only give a handle
-        # we verify the credentials via an actual operation
-        connect(options).inventory.list_feeds
-      rescue URI::InvalidComponentError
-        raise MiqException::MiqHostError, "Host '#{hostname}' is invalid"
-      rescue ::Hawkular::ConnectionException
-        raise MiqException::MiqUnreachableError, "Unable to connect to #{hostname}:#{port}"
-      rescue ::Hawkular::Exception => he
-        raise MiqException::MiqInvalidCredentialsError, 'Invalid credentials' if he.status_code == 401
-        raise MiqException::MiqHostError, 'Hawkular not found on host' if he.status_code == 404
-        raise MiqException::MiqCommunicationsError, he.message
-      rescue => err
-        $log.error(err)
-        raise MiqException::Error, 'Unable to verify credentials'
-      end
+      validate_connection(connect(options), options[:hostname], options[:port])
 
       true
     end
 
+    def self.validate_connection(connection, hostname = nil, port = nil)
```

Comment:
Mark this as `private`

Reply:
Why it should be private? Neither [VmWare](https://github.com/ManageIQ/manageiq-providers-vmware/blob/68f09bf0406e6e4992922157f3d94c0b493eb4c8/app/models/manageiq/providers/vmware/manager_auth_mixin.rb#L57) nor [OpenStack](https://github.com/ManageIQ/manageiq-providers-openstack/blob/bec9f82f8b56b6f2ae537a3d0343101b6984ccc8/app/models/manageiq/providers/openstack/manager_mixin.rb#L40) nor [Amazon](https://github.com/ManageIQ/manageiq-providers-amazon/blob/e46a7cb7f4edc1e9257dfd23789f1f88065c9016/app/models/manageiq/providers/amazon/manager_mixin.rb#L69) nor [Azure](https://github.com/ManageIQ/manageiq-providers-azure/blob/dbb5d39fe0b280b75e7b06d093b11ef6c0d42207/app/models/manageiq/providers/azure/manager_mixin.rb#L39) has these methods as private.

Reply:
@israel-hdez thanks very much for explanation! I didn't realized they are mixins. Cool, I'm making them private now! ;)

Reply:
@israel-hdez, ok, I've found out, the "old way" without queued validation is not working when this method is private (because it's called from instance via `verify_credentials`). The other method can be made private for sure.

Reply:
Private class methods are an anti-pattern. If we want to have this not be part of the interface then we should make it an instance method, or extract connection validation to another class (which I would prefer tbh).